sdk/identity/azure-identity/azure/identity/_exceptions.py

```diff
@@ -7,3 +7,7 @@
 
 class CredentialUnavailableError(ClientAuthenticationError):
     """The credential did not attempt to authenticate because required data or state is unavailable."""
+
+
+class AuthenticationRequiredError(CredentialUnavailableError):
```

Comment:
Why AuthenticationRequiredError is a subclass of CredentialUnavailableError?

It seems to me AuthenticationRequiredError is a HttpRequestError but CredentialUnavailableError may be not?

Reply:
Silent authentication can succeed or fail without sending a request. The question is really about the behavior of chained credentials: if a credential is configured to authenticate silently only, should a chain try its next credential when silent auth fails? I lean toward "yes", hence CredentialUnavailableError. What do you think?